src/collection.jl

```diff
@@ -9,9 +9,64 @@ end
 
 parse_name(filename, ext, ::Val{true}) = filename
 
+# utility function for generating filename
+function generate_filename(path, save_location, save_prefix, save_suffix, save_delim, ext)
+    # get the filename
+    filename = last(splitdir(path))
```

Comment:
Isn't it equivalent to
```suggestion
    filename = basename(path)
```
?

Reply:
Made the change, thank you!